File: 백준/Silver/1931. 회의실 배정/회의실 배정.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int[][] arr = new int[n][2];
        for (int i = 0; i < n; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            arr[i][0] = Integer.parseInt(st.nextToken());
            arr[i][1] = Integer.parseInt(st.nextToken());
        }
        
        Arrays.sort(arr, (arr1, arr2) -> {
            if (arr1[1] == arr2[1])
                return Integer.compare(arr1[0], arr2[0]);
            else return Integer.compare(arr1[1], arr2[1]);
        });
        int end = 0;
        int cnt = 0;
        for (int i = 0; i < n; i++)
            if (arr[i][0] >= end) {
                cnt++;
                end = arr[i][1];
            }
        System.out.print(cnt);
    }
}

File: 백준/Silver/1931. 회의실 배정/회의실 배정.scala
import java.io._

object Main {
    def main(args: Array[String]): Unit = {
        val br = new BufferedReader(new InputStreamReader(System.in));
        val n = br.readLine.toInt
        var arr = Vector[(Int, Int)]()
        for (i <- 1 to n) {
            val s = br.readLine.split(" ").map(_.toInt)
            arr = arr :+ (s(0), s(1))
        }
        
        print(arr.sortWith {case (t1, t2) => 
            if (t1._2 == t2._2) t1._1 < t2._1
            else t1._2 < t2._2}
            .foldLeft((0, (0, 0))){ case ((sum, t), cur) =>
                if (t._2 <= cur._1) sum + 1 -> cur
                else sum -> t
            }._1)
    }
}